lib/c-api/src/wasm_c_api/mod.rs

```diff
@@ -712,7 +712,13 @@ pub unsafe extern "C" fn wasm_func_new(
             num_rets
         ];
 
-        let _traps = callback(processed_args.as_ptr(), results.as_mut_ptr());
+        let trap = callback(processed_args.as_ptr(), results.as_mut_ptr());
+        dbg!(&trap);
+        if !trap.is_null() {
+            let trap: Box<wasm_trap_t> = Box::from_raw(trap);
+            dbg!("raising trap!");
+            RuntimeError::raise(Box::new(trap.inner));
```

Comment:
hmm, I don't know... I'm going to say probably not.  `repr(transparent)` works at the ABI level whereas in order to be passed to `RuntimeError::raise` you have to implement `std::error::Error` at the type level.  So with unsafe / C FFI everything would just work, but in safe Rust, I don't believe it would